.github/workflows/ci.yml

```
@@ -1,168 +1,257 @@
+            # Use keyword exclusion to avoid performance and integration tests that may not be properly marked
+            echo "Using keyword-based test exclusion for PR/feature branch"
+            PYTHONPATH=src pytest tests/ -n auto -v --cov=src --cov-report=xml --cov-report=html \
+              -k "not (workflow_execution_time or async_workflow_performance or async_workflow_integration)"
+          else
+            # Main branch: All tests with parallel execution
+            PYTHONPATH=src pytest tests/ -n auto -v --cov=src --cov-report=xml --cov-report=html
+          fi
+      
+      - name: Upload coverage
+        if: matrix.python-version == '3.10'
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)